cmd/panacead/cmd/genoracle.go

```diff
@@ -164,24 +164,32 @@ func setOracle(cmd *cobra.Command, genState *oracletypes.GenesisState) error {
 		return fmt.Errorf("failed to get oracle endpoint: %w", err)
 	}
 
-	commRateStr, err := cmd.Flags().GetString(flagOracleCommRate)
+	commMaxRateStr, err := cmd.Flags().GetString(flagOracleCommMaxRate)
 	if err != nil {
-		return fmt.Errorf("failed to get oracle commission rate: %w", err)
+		return fmt.Errorf("failed to get oracle commission max rate: %w", err)
 	}
 
-	commRate, err := sdk.NewDecFromStr(commRateStr)
+	commMaxRate, err := sdk.NewDecFromStr(commMaxRateStr)
 	if err != nil {
-		return fmt.Errorf("inavlid commission rate: %w", err)
+		return fmt.Errorf("inavlid commission max rate: %w", err)
 	}
 
-	commMaxRateStr, err := cmd.Flags().GetString(flagOracleCommMaxRate)
+	if commMaxRate.IsNegative() || commMaxRate.GT(sdk.OneDec()) {
+		return fmt.Errorf("oracle max commission rate should be between 0 and 1")
+	}
+
+	commRateStr, err := cmd.Flags().GetString(flagOracleCommRate)
 	if err != nil {
-		return fmt.Errorf("failed to get oracle commission max rate: %w", err)
+		return fmt.Errorf("failed to get oracle commission rate: %w", err)
 	}
 
-	commMaxRate, err := sdk.NewDecFromStr(commMaxRateStr)
+	commRate, err := sdk.NewDecFromStr(commRateStr)
 	if err != nil {
-		return fmt.Errorf("inavlid commission max rate: %w", err)
+		return fmt.Errorf("inavlid commission rate: %w", err)
+	}
+
+	if commRate.IsNegative() || commRate.GT(commMaxRate) {
+		return fmt.Errorf("oracle commission rate should be between 0 and commission max rate")
 	}
 
 	commMaxChangeRateStr, err := cmd.Flags().GetString(flagOracleCommMaxChangeRate)
```

Comment:
we need to check if the max change rate is greater than 0 and less or equal than max rate.

Reply:
Ah, Sorry. There is no need to check here. Because it is checked in `GenesisState.Validate()`. So, I remove all that checks like main branch.